File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;

class Solution {
    public int[] solution(int[] answers) {
        int l = answers.length;
        
        int[] first = {1, 2, 3, 4, 5};
        int[] second = {2, 1, 2, 3, 2, 4, 2, 5};
        int[] third = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        
        int fCount = 0;
        int sCount = 0;
        int tCount = 0;
        
        for (int i = 0; i < l; i++) {
            if (answers[i] == first[i % first.length]) {
                fCount++;    
            }
            if (answers[i] == second[i % second.length]) {
                sCount++;    
            }
            if (answers[i] == third[i % third.length]) {
                tCount++;    
            }
        }
        
        int maxScore = Math.max(fCount, Math.max(sCount, tCount));
        ArrayList<Integer> answer = new ArrayList<>();
        
        if (maxScore == fCount) answer.add(1);
        if (maxScore == sCount) answer.add(2);
        if (maxScore == tCount) answer.add(3);
        
        int[] result = new int[answer.size()];
        for (int i = 0; i < answer.size(); i++) {
            result[i] = answer.get(i);
            }
        return result;
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    
    vector<int> first = {1, 2, 3, 4, 5};
    vector<int> second = {2, 1, 2, 3, 2, 4, 2, 5};
    vector<int> third = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};

    int fCount = 0; 
    int sCount = 0;
    int tCount = 0; 

    for (int i = 0; i < answers.size(); i++) {
        if (answers[i] == first[i % first.size()]) {
            fCount++;
        }
        
        if (answers[i] == second[i % second.size()]) {
            sCount++;
        }
        
        if (answers[i] == third[i % third.size()]) {
            tCount++;
        }
        
    }
    int maxScore = max({fCount, sCount, tCount});  
        
    if (fCount == maxScore) answer.push_back(1);
    if (sCount == maxScore) answer.push_back(2);
    if (tCount == maxScore) answer.push_back(3);

    return answer;
}